### ISSUE TITLE
go() clears previously set waypoints

### ISSUE THREAD

**Issue by bemson** [bemson/salt#4]
The go() method clears previously set waypoints, when called internally or externally.

The following component function, demonstrates the issue when handled internally. The _in component does _not_ route the navigation to "b", as expected.

```
var page = new Flow({
  _in: function () {
    this.go('b');
    this.go();
  },
  _main: function () {
    console.log('fired A second!');
  },
  b: function () {
    console.log('fired B first!');
  }
});
page(); // -> fired A second!
```

The following function, demonstrates the issue when called externally. The statement does _not_ route the navigation to "b", as expected.

```
var page = new Flow({
  _in: function () {
    this.go('b');
    this.wait();
  },
  _main: function () {
    console.log('fired A second!');
  },
  b: function () {
    console.log('fired B first!');
  }
});
page();

Flow(page).go(); // -> fired A second!
```

The cause is that the staged waypoints are not committed after exiting the component function, and the go() method blindly sets the given waypoints, even if the waypoints array is empty; this is not the expected behavior.

This issue should be addressed in the next release.

**Comment by bemson** [comment]
Additional scenario demonstrates this issue when called externally:

```
var page = new Flow({
  _in: function () {
    this.go('b');
    this.wait();
  },
  _main: function () {
    console.log('fired A second!');
  },
  b: function () {
    console.log('fired B first!');
  },
  c: function () {
    console.log('fired C before A and B!');
  }
});
page();

Flow(page).go('c'); // -> fired C before A and B!
                    // -> fired A second!
```

The expected state executions (given this program) are C, B, and A. However, this example skips calling B, because the external call to go() contains a waypoint. Again, the internal staging needs to be committed in order to avoid this issue.

**Comment by bemson** [comment]
All three scenarios fixed in v0.2.1.2
